inc/themes/core.base/frontend/templates/showthread/showthread.twig

```diff
@@ -244,13 +244,27 @@
                     {% for post in threadedbits %}
                         {% if post.bitactive %}
                             <div class="threaded-mode__post threaded-mode__post--active" style="margin-left: {{ post.indentsize }}px;">
-                                <span class="threaded-mode__date">{{ post.postdate|raw }}</span>
-                                <span class="threaded-mode__author">{{ lang.by }} {{ post.profilelink|raw }}</span>
+                                <span class="threaded-mode__date">{{ post.postdate }}</span>
+                                <span class="threaded-mode__author">
+                                    {{ lang.by }}
+                                    {% if post.uid > 0 %}
+                                        <a href="{{ get_profile_link(post.uid) }}">{{ post.uid|format_name }}</a>
+                                    {% else %}
+                                        {{ post.username|default(lang.guest) }}
+                                    {% endif %}
+                                </span>
                             </div>
                         {% else %}
                             <div class="threaded-mode__post" style="margin-left: {{ post.indentsize }}px;">
-                                <span class="threaded-mode__date"><a href="showthread.php?tid={{ thread.tid }}&amp;pid={{ post.pid }}&amp;mode=threaded">{{ post.postdate|raw }}</a></span>
-                                <span class="threaded-mode__author">{{ lang.by }} {{ post.profilelink|raw }}</span>
+                                <span class="threaded-mode__date"><a href="showthread.php?tid={{ thread.tid }}&amp;pid={{ post.pid }}&amp;mode=threaded">{{ post.postdate }}</a></span>
```

Comment:
Shouldn't this use `get_post_link` (with a new parameter similar to `get_thread_link`) ?